.gitignore

```diff
@@ -1,34 +1,1208 @@
-# From https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore
-# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm
+# File created using '.gitignore Generator' for Visual Studio Code: https://bit.ly/vscode-gig
 
-## Directory-based project format
-.idea/
-# if you remove the above rule, at least ignore user-specific stuff:
-# .idea/workspace.xml
-# .idea/tasks.xml
-# and these sensitive or high-churn files:
-# .idea/dataSources.ids
-# .idea/dataSources.xml
-# .idea/sqlDataSources.xml
-# .idea/dynamic.xml
-
-## File-based project format
-*.ipr
+# Created by https://www.gitignore.io/api/visualstudiocode,macos,c,c++,code,code-java,database,dbeaver,eclipse,flask,git,go,helm,intellij,intellij+all,intellij+iml,java,java-web,jmeter,kotlin,linux,osx,pycharm,pycharm+all,pycharm+iml,pydev,python,visualstudio,web,windows
+# Edit at https://www.gitignore.io/?templates=visualstudiocode,macos,c,c++,code,code-java,database,dbeaver,eclipse,flask,git,go,helm,intellij,intellij+all,intellij+iml,java,java-web,jmeter,kotlin,linux,osx,pycharm,pycharm+all,pycharm+iml,pydev,python,visualstudio,web,windows
+
+### C ###
+# Prerequisites
+*.d
+
+# Object files
+*.o
+*.ko
+*.obj
+*.elf
+
+# Linker output
+*.ilk
+*.map
+*.exp
+
+# Precompiled Headers
+*.gch
+*.pch
+
+# Libraries
+*.lib
+*.a
+*.la
+*.lo
+
+# Shared objects (inc. Windows DLLs)
+*.dll
+*.so
+*.so.*
+*.dylib
+
+# Executables
+*.exe
+*.out
+*.app
+*.i*86
+*.x86_64
+*.hex
+
+# Debug files
+*.dSYM/
+*.su
+*.idb
+*.pdb
+
+# Kernel Module Compile Results
+*.mod*
+*.cmd
+.tmp_versions/
+modules.order
+Module.symvers
+Mkfile.old
+dkms.conf
+
+### C++ ###
+# Prerequisites
+
+# Compiled Object files
+*.slo
+
+# Precompiled Headers
+
+# Compiled Dynamic libraries
+
+# Fortran module files
+*.mod
+*.smod
+
+# Compiled Static libraries
+*.lai
+
+# Executables
+
+### Code ###
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
+
+### Code-Java ###
+# Language Support for Java(TM) by Red Hat extension for Visual Studio Code - https://marketplace.visualstudio.com/items?itemName=redhat.java
+
+.project
+.classpath
+factoryConfiguration.json
+
+### Database ###
+*.accdb
+*.db
+*.dbf
+*.mdb
+*.sqlite3
+
+### DBeaver ###
+# ide config file
+.dbeaver-data-sources*.xml
+
+### Eclipse ###
```

Comment:
The autogenerator automatically comments about plaform too. I have included .gitignore of some python and other language files too as we would be building clients in the future.